libafl/src/inputs/bytes.rs

```diff
@@ -146,4 +146,15 @@ mod test {
         let bytes = BytesInput::deserialize_dynamic(&buf).unwrap().unwrap();
         assert_eq!(bytes.target_bytes().as_slice(), b"hello");
     }
+
+    #[test]
+    fn failed_deserialize_from_nop() {
+        // note that NopInput implements HasTargetBytes, but because we have not submitted the
+        // conversion BytesInput cannot be converted from NopInput
+
+        let nop = NopInput {};
+        let mut buf = Vec::new();
+        nop.serialize_dynamic(&mut buf).unwrap();
+        assert!(BytesInput::deserialize_dynamic(&buf).unwrap().is_none());
+    }
```

Comment:
See comments here: if the deserialisation for this type cannot occur, it is skipped and not deserialised.